plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -481,7 +567,12 @@ private static ConnectorPageSource createParquetPageSource(
             // Map by name for a migrated table
             boolean mapByName = parquetIdToField.isEmpty();
 
-            List<org.apache.parquet.schema.Type> parquetFields = regularColumns.stream()
+            Optional<ReaderColumns> columnProjections = projectColumns(regularColumns);
+            List<IcebergColumnHandle> readColumns = columnProjections
+                    .map(readerColumns -> (List<IcebergColumnHandle>) readerColumns.get().stream().map(IcebergColumnHandle.class::cast).collect(toImmutableList()))
+                    .orElse(regularColumns);
+
+            List<org.apache.parquet.schema.Type> parquetFields = readColumns.stream()
```

Comment:
are we intentionally not pushing predicates into parquet reader? (line 591)

Reply:
Probably just because it's relatively new. The change allowing for passing a Predicate to ParquetReader was just merged in August. If it's useful here, I'd rather do it in followup, since it's related to all the column index changes to the parquet reader and I don't really understand those yet. https://github.com/trinodb/trino/commit/6eb42f2aeca9a07a53249ae4bd56c13c8f002335

Reply:
I was mainly thinking of the pushdown in rowgroups (ORC stripe-equivalent, even before newly added code). but look like it's missing in hive connector too. cc @JamesRTaylor

Reply:
i am not sure i understand the conclusion. should we have a follow-up issue for this?

Reply:
[exclusion](https://github.com/trinodb/trino/blob/master/plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetPageSourceFactory.java#L201) of nested predicates was added initially when only ORC support was added for dereference pushdown. However, seems like it didn't change in #3396 while adding parquet support. I'm not sure if that was intentional. but you're right that this deserves a separate issue. #9928

For iceberg connector, I don't see corresponding changes like #3396 here for [propagating projected layout](https://github.com/trinodb/trino/pull/8129#discussion_r745227296), so assumed that we're tackling it separately.

Reply:
#9931